api/chart/chart.html

```diff
@@ -229,6 +229,9 @@
         var deviceIdStr = "";
         if($('#devices').find(':selected').text().length > 0){
             deviceIdStr = "&deviceId=" + $('#devices').find(':selected').text();
+        } else {
+          // due to async nature of devices array population, on first page load url variable at line 236 would get empty string insted of first device name, if not specified, hence:
+            deviceIdStr = "&deviceId=RaspPi-GasMonitor02";
```

Comment:
I understand the motivation behind this, as you mentioned in the comment, however I wonder if there's a more future proof solution. In other words, what if we end up changing the device names? This won't work at that point.

Reply:
One way to achieve this, is to have a listener on the drop down selector with the devices, and when it finishes getting the devices back from the server, then it attempts to load the data in the chart for the first deviceId, regardless of what it is named.